sklearn/neighbors/base.py

```diff
@@ -258,6 +258,19 @@ def _fit(self, X):
         else:
             raise ValueError("algorithm = '%s' not recognized"
                              % self.algorithm)
+
+        if self.n_neighbors is not None:
```

Comment:
`n_neighbors` doesn't exist in all instances of `NeighborsBase` (i.e. where radius neighbors are intended).

I also don't understand why anyone would set `n_neighbors` to None. Or why we want to handle things like strings...